contentcuration/contentcuration/utils/publish.py

```diff
@@ -708,18 +728,40 @@ def process_image_strings(content, zf, channel_id):
                     logging.warning("NOTE: the following error would have been swallowed silently in production")
                     raise
 
-            image_name = "images/{}.{}".format(checksum, ext[1:])
-            if image_name not in zf.namelist():
-                with storage.open(ccmodels.generate_object_storage_name(checksum, filename), 'rb') as imgfile:
-                    write_to_zipfile(image_name, imgfile.read(), zf)
-
-            # Add resizing data
-            if img_match.group(2) and img_match.group(3):
-                image_data = {'name': img_match.group(1)}
-                image_data.update({'width': float(img_match.group(2))})
-                image_data.update({'height': float(img_match.group(3))})
-                image_list.append(image_data)
-            content = content.replace(match.group(1), img_match.group(1))
+            original_image_name = "images/{}.{}".format(checksum, ext[1:])
+            with storage.open(ccmodels.generate_object_storage_name(checksum, filename), 'rb') as imgfile:
+                original_content = imgfile.read()
+
+                if img_match.group(2) and img_match.group(3):
+                    width, height = float(img_match.group(2)), float(img_match.group(3))
+                    resized_content = resize_image(original_content, width, height)
+
+                    if resized_content:
+                        resized_checksum = get_resized_image_checksum(resized_content)
+                        new_image_name = "images/{}.{}".format(resized_checksum, ext[1:])
+
+                        if new_image_name not in zf.namelist():
+                            write_to_zipfile(new_image_name, resized_content, zf)
+
+                        original_img_ref = match.group(1)
+                        new_img_ref = original_img_ref.replace(filename, f"{resized_checksum}{ext}")
+                        new_img_match = re.search(r'(.+/images/[^\s]+)(?:\s=([0-9\.]+)x([0-9\.]+))*', new_img_ref)
+
+                        # Add resizing data
+                        image_data = {'name': new_img_match.group(1)}
+                        image_data.update({'width': width})
+                        image_data.update({'height': height})
+                        image_list.append(image_data)
+                        content = content.replace(original_img_ref, new_img_match.group(1))
+                    else:
+                        logging.warning(f"Failed to resize image {filename}. Using original image.")
+                        if original_image_name not in zf.namelist():
+                            write_to_zipfile(original_image_name, original_content, zf)
+                        content = content.replace(match.group(1), img_match.group(1))
+                else:
+                    if original_image_name not in zf.namelist():
```

Comment:
Seems in this high level path, if it results in `False`, then `original_content` goes unused?